frontend/src/metabase/lib/colors/palette.ts

```diff
@@ -80,17 +80,27 @@ const aliases: Record<string, (palette: ColorPalette) => string> = {
   "accent7-dark": palette => shade(color(`accent7`, palette)),
 };
 
-export const color = (color: string, palette = colors) => {
+export function color(
+  colorName: keyof ColorPalette,
+  palette?: ColorPalette,
+): string;
+export function color(color: string, palette?: ColorPalette): string;
+export function color(color: any, palette: ColorPalette = colors) {
```

Comment:
I tested using union type i.e. `color: string | keyof ColorPalette`, but color name autocompletion doesn't work.

This uses [function overloading](https://www.typescriptlang.org/docs/handbook/2/functions.html#function-overloads) instead.